test/HeapProfilerAgent.js

```diff
@@ -188,23 +184,23 @@ describe('HeapProfilerClient', function() {
 });
 
 function initializeProfiler(done) {
-  launcher.runPeriodicConsoleLog(true, function(childProcess, client) {
-    var injectorClient = new InjectorClient({}, client);
-    debuggerClient = client;
-    heapProfilerClient = new HeapProfilerClient({}, debuggerClient);
-    heapProfilerAgent = new HeapProfilerAgent(
-      {},
-      debuggerClient,
-      injectorClient,
-      frontendClient,
-      heapProfilerClient);
+  launcher.runPeriodicConsoleLog(true, function(childProcess, session) {
+    debuggerClient = session.debuggerClient;
+    frontendClient = session.frontendClient;
+
+    var injectorClient = new InjectorClient({}, session);
+    session.injectorClient = injectorClient;
+
+    heapProfilerClient = new HeapProfilerClient({}, session);
+    session.heapProfilerClient = heapProfilerClient;
+
+    heapProfilerAgent = new HeapProfilerAgent({}, session);
```

Comment:
If you look at `lib/session.js`, you can see that the session object provides these objects:

```
+  this.debuggerClient = new DebuggerClient(debuggerPort);
+  this.frontendClient = new FrontendClient(wsConnection);
+  this.injectorClient = new InjectorClient(config, this);
+  this.consoleClient = new ConsoleClient(config, this);
+  this.heapProfilerClient = new HeapProfilerClient(config, this);
+  this.scriptManager = new ScriptManager(config, this);
+  this.breakEventHandler = new BreakEventHandler(config, this);
+  this.frontendCommandHandler = new FrontendCommandHandler(config, this);
```

I am saying that the `SessionStub` used in tests should provide all of these objects too. So that you don't have to create e.g. `heapProfilerClient` manually (like in the test here), but instead you can rely on the fact that `SessionStub` provides the same set of objects as the real `Session`.